### PR TITLE
Add CI

### PR DESCRIPTION
**Pull request by ahmetsait**
This PR tries to add CI using GitHub Actions. The current test coverage is quite poor but will be better as we add tests.
On a related note: Not sure how should we use png/bmp test suits, downloading them on every CI invokation seems wasteful and they have their own varying licenses as well. Should probably check CI docs for caching and what not.

### CODE REVIEW

**Comment by p0nce** [comment]
You can probably upload a file for the CI then unzip it as a CI action. 
HUGE +1 :)

File: src/wg/format/png/reader.d [hunk 1]
```diff
@@ -5,7 +5,9 @@ import wg.format.png.types;
 import wg.util.util: Result, asArrayOf;
 import wg.image.imagebuffer;
 
-// TODO: 
+import etc.c.zlib : inflateInit, inflate, inflateEnd, z_stream, z_streamp;
+
+// TODO:
 // 1. Make CommonMetadata using pHYs chunk data
 // 2. Make PngChunks smarter so they keep their allocator and chunks can easily be added and removed
 // 3. Check sRGB, gAMA and cHRM chunks and return the proper format string using them
```
Comment (igor84) [inline src/wg/format/png/reader.d]: I just realized the third TODO item here is actually done so it can be deleted.

Comment (ahmetsait) [reply src/wg/format/png/reader.d]: Feels a bit unrelated but if you say so..?

**Comment by rikkimax** [comment]
Nobody saying anything is wrong, merging.

**Pull request merged**